### DISCUSSION TITLE
Unable to resolve dependency

### DISCUSSION THREAD

**Discussion by jbuijgers** [chocolatey/home#334] [Q&A]
Creating a new package, got the following XML config under `package.metadata`.
```xml
<dependencies>
    <dependency id="chocolatey-core.extension" version="1.4" />
    <dependency id="dotnet-7.0-desktopruntime" version="7.0.20" />
</dependencies>
```

When I run `choco install my-package --source .`, I get `Unable to resolve dependency 'dotnet-7.0-desktopruntime'.`. I'm a bit confused, I'd expect this to work. Anyone got ideas on how to solve this?

**Comment by jbuijgers** [comment]
Got an answer from the Discord channel. Adding chocolatey as a source worked for me. So the command became `choco install my-package --source .,chocolatey`. Also, removed the 'chocolatey-core.extension' dependency, since it is added by default.

**Reply by corbob** [reply]
To expand on this: `chocolatey-core.extension` is not installed by default, but many packages on the community repository include it as a dependency. If you're using functions from the extension, you should definitely include it as a dependency.
